test/function/severity.cpp

```diff
@@ -122,4 +122,24 @@ namespace {
     EXPECT_EQ( result, -1 );
   }
 
-}
+  TEST( GetSeverityEnumFromBuffer, NoSuchSeverity ) {
+    int result;
+
+    result = stumpless_get_severity_enum_from_buffer( "an_invalid_severity", 10 );
+    EXPECT_EQ( result, -1 );
+  }
+
+  TEST( GetSeverityEnumFromBuffer, InvalidMemSeverity ) {
+    int result;
+    void * (*set_malloc_result)(size_t);
+    set_malloc_result = stumpless_set_malloc( MALLOC_FAIL );
+    ASSERT_NOT_NULL( set_malloc_result );
+
+    result = stumpless_get_severity_enum_from_buffer( "info", sizeof("info") );
+    EXPECT_EQ( result, -1 );
```

Comment:
There should also be an `EXPECT_ERROR_ID_EQ` check here to make sure that the error code is what is expected in this case. It looks like that's also missing in the test that you based this one on, so feel free to add it there as well.